src/cmap/conn/wire/message.rs

```diff
@@ -36,6 +37,12 @@ impl Message {
                 .insert("$readPreference", read_pref.into_document());
         };
 
+        if let Some(server_api) = server_api {
+            if server_api.applies_to_command(command.name.as_str()) {
+                server_api.append_to_command(&mut command.body);
+            }
```

Comment:
With the alternative API in `ServerApi`:
```suggestion
            server_api.append_to_command(&mut command);
```